pkg/imageselector/name/name.go

```diff
@@ -0,0 +1,78 @@
+package name
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/acorn-io/runtime/pkg/imagepattern"
+	"github.com/google/go-containerregistry/pkg/name"
+)
+
+func ImageCovered(image name.Reference, digest string, patterns []string) bool {
+	for _, pattern := range patterns {
+		// empty pattern? skip (should've been caught by IAR validation already)
+		if strings.TrimSpace(pattern) == "" {
+			continue
+		}
+
+		// not a pattern? must be exact match then.
+		if !imagepattern.IsImagePattern(pattern) {
+			if strings.TrimSuffix(image.Name(), ":") != pattern && digest != pattern {
```

Comment:
I've seen things.. 
Actually there are cases where the image reference gets parsed as a repository-only reference, i.e. without a tag or digest attached. Since there's [no validation that the identifier is non-empty in ggcr,](https://github.com/google/go-containerregistry/blob/main/pkg/name/tag.go#L53-L55) in those cases we'd end up with a `:` followed by nothing.

Reply:
Oh yeah, I didn't think about not having `latest` as a default tag. That makes sense